### PR TITLE
Editorial feedback for Expression syntax

### PR DESCRIPTION
**Pull request by carolxob**
### Description
Editorial feedback for Expression syntax

### Issues Resolved
Addresses feedback in #1947 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _data-prepper/expression-syntax.md [hunk 1]
```diff
@@ -4,7 +4,9 @@ title: Expression syntax
 nav_order: 12
 ---
 
-# Expression syntax
+# Expression syntax 
+
+This page discusses expression syntax.
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:9]:
```suggestion
The following sections provide information about expression syntax.
```

File: _data-prepper/expression-syntax.md [hunk 2]
```diff
@@ -81,9 +86,9 @@ Used to test whether two values are equivalent.
 3.14 != /status_code
 {1, 2} == /event/set_property
 ```
-## Using equality operators to check JSON pointer 
+## Using equality operators to check for JSON Pointer
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:89]:
```suggestion
## Using equality operators to check for a JSON Pointer 
```

File: _data-prepper/expression-syntax.md [hunk 3]
```diff
@@ -120,20 +125,22 @@ not /status_code in {200, 202}
...
-- Integer _(Supports values from &minus;2,147,483,648 to 2,147,483,647)_
-- Boolean _(Supports true or false)_
-- JSON Pointer _(See JSON Pointer section for details)_
-- String _(Supports valid Java strings)_
-- Null _(Supports null check to see whether a JSON pointer exists)_
+A literal is a fundamental value that has no children.
+- Float: Supports values from 3.40282347 &times; 10<sup>38</sup> to 1.40239846 &times; 10<sup>&minus;45</sup>.
+- Integer: Supports values from &minus;2,147,483,648 to 2,147,483,647.
+- Boolean: Supports true or false.
+- JSON Pointer: See the [JSON Pointer](#json-pointer) section below for details.
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:135]:
```suggestion
- JSON Pointer: See the [JSON Pointer](#json-pointer) section for details.
```

File: _data-prepper/expression-syntax.md [hunk 4]
```diff
@@ -120,20 +125,22 @@ not /status_code in {200, 202}
...
+
 ### Literal
-A fundamental value that has no children.
-- Float _(Supports values from 3.40282347 &times; 10<sup>38</sup> to 1.40239846 &times; 10<sup>&minus;45)</sup>_
-- Integer _(Supports values from &minus;2,147,483,648 to 2,147,483,647)_
-- Boolean _(Supports true or false)_
-- JSON Pointer _(See JSON Pointer section for details)_
-- String _(Supports valid Java strings)_
-- Null _(Supports null check to see whether a JSON pointer exists)_
+A literal is a fundamental value that has no children.
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:131]:
```suggestion
A literal is a fundamental value that has no children:
```

File: _data-prepper/expression-syntax.md [hunk 5]
```diff
@@ -81,9 +86,9 @@ Used to test whether two values are equivalent.
 3.14 != /status_code
 {1, 2} == /event/set_property
 ```
-## Using equality operators to check JSON pointer 
+## Using equality operators to check for JSON Pointer 
 
-Equality operators can also be used to check whether a JSON Pointer exists by comparing with `null`.
+Equality operators can also be used to check whether a JSON Pointer exists by comparing the JSON Pointer with `null`.
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:91]: This is slightly confusing. How can we use something for comparison if we don't know it exists yet?

File: _data-prepper/expression-syntax.md [hunk 6]
```diff
@@ -120,20 +125,22 @@ not /status_code in {200, 202}
...
+- Float: Supports values from 3.40282347 &times; 10<sup>38</sup> to 1.40239846 &times; 10<sup>&minus;45</sup>.
+- Integer: Supports values from &minus;2,147,483,648 to 2,147,483,647.
+- Boolean: Supports true or false.
+- JSON Pointer: See the [JSON Pointer](#json-pointer) section below for details.
+- String: Supports valid Java strings.
+- Null: Supports null check to see whether a JSON pointer exists.
 
 ### Expression string
-The string that will be parsed for evaluation. Expression string is the highest level of a Data Prepper expression. Only supports one expression string resulting in a return value. An _expression string_ is not the same as an _expression_.
+Expression string is the highest-priority in a Data Prepper expression and only supports one expression string resulting in a return value. An _expression string_ is not the same as an _expression_.
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:140]:
```suggestion
An expression string takes the highest priority in a Data Prepper expression and only supports one expression string resulting in a return value. An _expression string_ is not the same as an _expression_.
```

File: _data-prepper/expression-syntax.md [hunk 7]
```diff
@@ -120,20 +125,22 @@ not /status_code in {200, 202}
...
 
 ### Statement
-The highest level component of the expression string.
+The highest-priority component of the expression string.
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:143]:
```suggestion
A statement is the highest-priority component of an expression string.
```

File: _data-prepper/expression-syntax.md [hunk 8]
```diff
@@ -145,13 +152,14 @@ A generic component that contains a _Primary_ or an _Operator_. Expressions may
 - _Literal_
 
 ### Operator
-Hard coded token that identifies the operation used in an _expression_.
+Hardcoded token that identifies the operation used in an _expression_.
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:155]:
```suggestion
An operator is a hardcoded token that identifies the operation used in an _expression_.
```

File: _data-prepper/expression-syntax.md [hunk 9]
```diff
@@ -120,20 +125,22 @@ not /status_code in {200, 202}
...
 ### Expression string
-The string that will be parsed for evaluation. Expression string is the highest level of a Data Prepper expression. Only supports one expression string resulting in a return value. An _expression string_ is not the same as an _expression_.
+Expression string is the highest-priority in a Data Prepper expression and only supports one expression string resulting in a return value. An _expression string_ is not the same as an _expression_.
 
 ### Statement
-The highest level component of the expression string.
+The highest-priority component of the expression string.
 
 ### Expression
 A generic component that contains a _Primary_ or an _Operator_. Expressions may contain expressions. An expressions imminent children can contains 0-1 _Operators_.
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:146]:
```suggestion
An expression is a generic component that contains a _Primary_ or an _Operator_. Expressions may contain expressions. An expression's imminent children can contain 0–1 _Operators_.
```

File: _data-prepper/expression-syntax.md [hunk 10]
```diff
@@ -145,13 +152,14 @@ A generic component that contains a _Primary_ or an _Operator_. Expressions may
 - _Literal_
 
 ### Operator
-Hard coded token that identifies the operation used in an _expression_.
+Hardcoded token that identifies the operation used in an _expression_.
 
 ### JSON pointer
-A Literal used to reference a value within the Event provided as context for the _Expression String_. JSON Pointers are identified by a 
+JSON Pointer is a literal used to reference a value within the event and provided as context for the _Expression string_. JSON Pointers are identified by a
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:158]:
```suggestion
A JSON Pointer is a literal used to reference a value within an event and provided as context for an _expression string_. JSON Pointers are identified by a 
```

File: _data-prepper/expression-syntax.md [hunk 11]
```diff
@@ -145,13 +152,14 @@ A generic component that contains a _Primary_ or an _Operator_. Expressions may
 - _Literal_
 
 ### Operator
-Hard coded token that identifies the operation used in an _expression_.
+Hardcoded token that identifies the operation used in an _expression_.
 
 ### JSON pointer
-A Literal used to reference a value within the Event provided as context for the _Expression String_. JSON Pointers are identified by a 
+JSON Pointer is a literal used to reference a value within the event and provided as context for the _Expression string_. JSON Pointers are identified by a 
 leading `/` containing alphanumeric character or underscores, delimited by `/`. JSON Pointers can use an extended character set if wrapped
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:159]:
```suggestion
leading `/` containing alphanumeric characters or underscores, delimited by `/`. JSON Pointers can use an extended character set if wrapped 
```

File: _data-prepper/expression-syntax.md [hunk 12]
```diff
@@ -145,13 +152,14 @@ A generic component that contains a _Primary_ or an _Operator_. Expressions may
...
-Hard coded token that identifies the operation used in an _expression_.
+Hardcoded token that identifies the operation used in an _expression_.
 
 ### JSON pointer
-A Literal used to reference a value within the Event provided as context for the _Expression String_. JSON Pointers are identified by a 
+JSON Pointer is a literal used to reference a value within the event and provided as context for the _Expression string_. JSON Pointers are identified by a 
 leading `/` containing alphanumeric character or underscores, delimited by `/`. JSON Pointers can use an extended character set if wrapped 
-in double quotes (`"`) using the escape character `\`. Note, JSON Pointer require `~` and `/` that should be used as part of the path and 
-not a delimiter to be escaped.
+in double quotes (`"`) using the escape character `\`. Note that JSON Pointers require `~` and `/` characters, which should be used as part of the path and not a delimiter needs to be escaped.
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:160]: "...and not as a delimiter that needs to be escaped."?

File: _data-prepper/expression-syntax.md [hunk 13]
```diff
@@ -145,13 +152,14 @@ A generic component that contains a _Primary_ or an _Operator_. Expressions may
...
+
+See the following examples of JSON pointers:
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:162]:
```suggestion
The following are examples of JSON Pointers:
```

File: _data-prepper/expression-syntax.md [hunk 14]
```diff
@@ -161,31 +169,37 @@ not a delimiter to be escaped.
 /\w+(/\w+)*
 ```
 
-##### Shorthand example
+#### Shorthand example
+
+See the following example of shorthand.
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:174]:
```suggestion
The following is an example of shorthand:
```

File: _data-prepper/expression-syntax.md [hunk 15]
```diff
@@ -161,31 +169,37 @@ not a delimiter to be escaped.
...
+
 ```
 /Hello/World/0
 ```
 
-##### Escaped syntax example
+#### Escaped syntax example
+
+See the following example of escaped syntax.
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:182]:
```suggestion
The following is an example of escaped syntax:
```

File: _data-prepper/expression-syntax.md [hunk 16]
```diff
@@ -161,31 +169,37 @@ not a delimiter to be escaped.
 /\w+(/\w+)*
 ```
 
-##### Shorthand example
+#### Shorthand example
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:172]:
```suggestion
#### Example of shorthand
```

File: _data-prepper/expression-syntax.md [hunk 17]
```diff
@@ -161,31 +169,37 @@ not a delimiter to be escaped.
...
+
+See the following example of shorthand.
+
 ```
 /Hello/World/0
 ```
 
-##### Escaped syntax example
+#### Escaped syntax example
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:180]:
```suggestion
#### Example of escaped syntax
```

File: _data-prepper/expression-syntax.md [hunk 18]
```diff
@@ -161,31 +169,37 @@ not a delimiter to be escaped.
...
+
+See the following example of escaped syntax.
 ```
 "/<Valid String Characters | Escaped Character>(/<Valid String Characters | Escaped Character>)*"
 ```
 
 #### Escaped example
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:187]:
```suggestion
#### Example of an escaped JSON Pointer
```

File: _data-prepper/expression-syntax.md [hunk 19]
```diff
@@ -161,31 +169,37 @@ not a delimiter to be escaped.
...
+
+See the following example of an escaped JSON Pointer.
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:189]:
```suggestion
The following is an example of an escaped JSON Pointer:
```

File: _data-prepper/expression-syntax.md [hunk 20]
```diff
@@ -161,31 +169,37 @@ not a delimiter to be escaped.
...
-White space is **optional** surrounding Relational Operators, Regex Equality Operators, Equality Operators and commas.
-White space is **required** surrounding Set Initializers, Priority Expressions, Set Operators, and Conditional Expressions.
 
-### Reference table
+White space is **optional** surrounding relational operators, regex equality operators, equality operators, and commas.
+White space is **required** surrounding set initializers, priority expressions, set operators, and conditional expressions.
+
 
-| Operator             | Description              | White Space Required | ✅ Valid Examples                                               | ❌ Invalid Examples                    |
+| Operator             | Description              | White space Required | ✅ Valid examples                                               | ❌ Invalid examples                    |
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:202]:
```suggestion
| Operator             | Description              | White space required | ✅ Valid examples                                               | ❌ Invalid examples                    |
```

File: _data-prepper/expression-syntax.md [hunk 21]
```diff
@@ -161,31 +169,37 @@ not a delimiter to be escaped.
...
 |----------------------|--------------------------|----------------------|----------------------------------------------------------------|---------------------------------------|
 | `{}`                 | Set Initializer          | Yes                  | `/status in {200}`                                             | `/status in{200}`                     |
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:204]:
```suggestion
| `{}`                 | Set initializer          | Yes                  | `/status in {200}`                                             | `/status in{200}`                     |
```

File: _data-prepper/expression-syntax.md [hunk 22]
```diff
@@ -161,31 +169,37 @@ not a delimiter to be escaped.
...
 | `()`                 | Priority Expression      | Yes                  | `/a==(/b==200)`<br>`/a in ({200})`                             | `/status in({200})`                   |
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:205]:
```suggestion
| `()`                 | Priority expression      | Yes                  | `/a==(/b==200)`<br>`/a in ({200})`                             | `/status in({200})`                   |
```

File: _data-prepper/expression-syntax.md [hunk 23]
```diff
@@ -145,13 +152,14 @@ A generic component that contains a _Primary_ or an _Operator_. Expressions may
 - _Literal_
 
 ### Operator
-Hard coded token that identifies the operation used in an _expression_.
+Hardcoded token that identifies the operation used in an _expression_.
 
 ### JSON pointer
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:157]:
```suggestion
### JSON Pointer
```

File: _data-prepper/expression-syntax.md [hunk 24]
```diff
@@ -93,14 +98,14 @@ null == <JSON Pointer>
...
-### Examples
+### Example
 ```
 /response == null
 null != /response
 ```
 
 #### Conditional expression
-Used to chain together multiple expressions and/or values.
+Conditional expression is used to chain together multiple expressions and/or values.
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:108]:
```suggestion
A conditional expression is used to chain together multiple expressions and/or values.
```

File: _data-prepper/expression-syntax.md [hunk 25]
```diff
@@ -41,17 +46,17 @@ Defines a set or term and/or expressions.
 ```
 
 ## Priority expression
-Identifies an expression that will be evaluated at the highest priority level. Priority expression must contain an
-expression or value, empty parentheses are not supported.
 
-### Examples
+Priority expression identifies an expression that will be evaluated at the highest priority level. Priority expression must contain an expression or value; empty parentheses are not supported.
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:50]:
```suggestion
A priority expression identifies an expression that will be evaluated at the highest priority level. A priority expression must contain an expression or value; empty parentheses are not supported.
```

File: _data-prepper/expression-syntax.md [hunk 26]
```diff
@@ -22,10 +24,13 @@ Operators are listed in order of precedence (top to bottom, left to right).
 Reserved symbol set: `^`, `*`, `/`, `%`, `+`, `-`, `xor`, `=`, `+=`, `-=`, `*=`, `/=`, `%=`, `++`, `--`, `${<text>}`
 
 ## Set initializer
-Defines a set or term and/or expressions.
+
+The set initializer defines a set or term and/or expressions.
 
 ### Examples
 
+See the following examples of set initializer.
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:32]:
```suggestion
The following are examples of set initializer syntax.
```

**Review by natebower — COMMENTED** [review]
@carolxob Please see my second round of changes and comments and let me know if you have any questions. Thanks!

File: _data-prepper/expression-syntax.md [hunk 27]
```diff
@@ -22,10 +24,13 @@ Operators are listed in order of precedence (top to bottom, left to right).
 Reserved symbol set: `^`, `*`, `/`, `%`, `+`, `-`, `xor`, `=`, `+=`, `-=`, `*=`, `/=`, `%=`, `++`, `--`, `${<text>}`
 
 ## Set initializer
-Defines a set or term and/or expressions.
+
+The set initializer defines a set or term and/or expressions.
```
[the remark this answers is not in the document]
Comment (carolxob) [reply _data-prepper/expression-syntax.md:28]: Confirmed, this should not be "A set initializer."

**Review by Naarcha-AWS — APPROVED** [review]

**Comment by carolxob** [comment]
@natebower Can you please review one more time? I experienced some technical issues with Git and want to make sure they're visible here.

File: _data-prepper/expression-syntax.md [hunk 28]
```diff
@@ -145,13 +152,14 @@ A generic component that contains a _Primary_ or an _Operator_. Expressions may
 - _Literal_
 
 ### Operator
-Hard coded token that identifies the operation used in an _expression_.
+An operator is a hardcoded token that identifies the operation used in an _expression_.
+
+### JSON Pointer
+A JSON Pointer is a literal used to reference a value within an event and provided as context for an _expression string_. JSON Pointers are identified by a 
+leading `/` containing alphanumeric characters or underscores, delimited by `/`. JSON Pointers can use an extended character set if wrapped 
+in double quotes (`"`) using the escape character `\`. Note that JSON Pointers require `~` and `/` characters, which should be used as part of the path and not a delimiter needs to be escaped.
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:160]:
```suggestion
in double quotes (`"`) using the escape character `\`. Note that JSON Pointers require `~` and `/` characters, which should be used as part of the path and not as a delimiter that needs to be escaped.
```

File: _data-prepper/expression-syntax.md [hunk 29]
```diff
@@ -161,34 +169,40 @@ not a delimiter to be escaped.
...
 |----------------------|--------------------------|----------------------|----------------------------------------------------------------|---------------------------------------|
-| `{}`                 | Set Initializer          | Yes                  | `/status in {200}`                                             | `/status in{200}`                     |
-| `()`                 | Priority Expression      | Yes                  | `/a==(/b==200)`<br>`/a in ({200})`                             | `/status in({200})`                   |
+| `{}`                 | Set initializer          | Yes                  | `/status in {200}`                                             | `/status in{200}`                     |
+| `()`                 | Priority expression      | Yes                  | `/a==(/b==200)`<br>`/a in ({200})`                             | `/status in({200})`                   |
 | `in`, `not in`       | Set Operators            | Yes                  | `/a in {200}`<br>`/a not in {400}`                             | `/a in{200, 202}`<br>`/a not in{400}` |
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:206]:
```suggestion
| `in`, `not in`       | Set operators            | Yes                  | `/a in {200}`<br>`/a not in {400}`                             | `/a in{200, 202}`<br>`/a not in{400}` |
```

File: _data-prepper/expression-syntax.md [hunk 30]
```diff
@@ -161,34 +169,40 @@ not a delimiter to be escaped.
...
 | `<`, `<=`, `>`, `>=` | Relational Operators     | No                   | `/status < 300`<br>`/status>=300`                              |                                       |
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:207]:
```suggestion
| `<`, `<=`, `>`, `>=` | Relational operators     | No                   | `/status < 300`<br>`/status>=300`                              |                                       |
```

File: _data-prepper/expression-syntax.md [hunk 31]
```diff
@@ -161,34 +169,40 @@ not a delimiter to be escaped.
...
 | `=~`, `!~`           | Regex Equality Operators | No                   | `/msg =~ "^\w*$"`<br>`/msg=~"^\w*$"`                           |                                       |
```
Comment (natebower) [inline _data-prepper/expression-syntax.md:208]:
```suggestion
| `=~`, `!~`           | Regex equality operators | No                   | `/msg =~ "^\w*$"`<br>`/msg=~"^\w*$"`                           |                                       |
```

**Review by natebower — APPROVED** [review]
Approved with these four commits.

**Pull request merged**